Update field properties (#286)
meilisearch/documentation: guides/advanced_guides/field_properties.md

# Field Properties

By default, every field in a document is **searchable** and **displayed**. These properties can be modified in the [settings](/references/settings.md).

## Searchable Fields

The **values** of the fields whose attributes are added to the [searchable-attributes list](/references/searchable_attributes.md) are **searched for matching query words**.

Their content is used by MeiliSearch to assess the relevancy of a document.

> **By default, all field attributes are added to the searchable-attributes list**. If a new document is ingested and one of its fields is missing from other documents, the latter will be automatically added to the searchable-attributes list. [This behavior can be changed](/references/accept_new_fields.md).

Reviewer: ```suggestion
**By default, all field attributes are added to the searchable-attributes list**. If a new document is ingested and one of its fields was never present in any other document, the latter will be automatically added to the searchable-attributes list. [This behavior can be changed](/references/accept_new_fields.md).
```

Therefore, if a field attribute is not in the searchable-attributes list, the field will be ignored from searches but will remain stored in the server. This list can be restricted to a selected set of attributes in the settings.

#### Example

Suppose you manage a database that contains information about movies. By adding the following settings, the fields `uid`, `movie_id`, `title`, `description`, `poster`, `release_date` and `rank` will be searched.

```bash
$ curl \
  -X POST 'http://localhost:7700/indexes/movies/settings' \
  --data '{
      "searchableAttributes": [
          "uid",
          "movie_id",
          "title",
          "description",
          "poster",
          "release_date",
          "rank"
      ]
  }'
```

## Displayed Fields

The fields whose attributes are added to the [displayed-attributes list](/references/displayed_attributes.md) are **displayed in each matching document**.

Documents returned upon search contain only displayed fields.

> **By default, all field attributes are added to the displayed-attributes list**. If a new document is ingested and one of its fields is missing from other documents, the latter will be automatically added to the displayed-attributes list. [This behavior can be changed](/references/accept_new_fields.md).

Reviewer: ```suggestion
**By default, all field attributes are added to the displayed-attributes list**. If a new document is ingested and one of its fields was never present in any other document, the latter will be automatically added to the displayed-attributes list. [This behavior can be changed](/references/accept_new_fields.md).
```

Author: More clear, thx!

> Therefore, if a field attribute is not in the displayed-attribute list, the field won't be added to the documents but will remain stored in the server. This list can be restricted to a selected set of attributes in the settings.

Reviewer: ```suggestion
Therefore, if a field attribute is not in the displayed-attribute list, the field won't be added to the documents returned upon search. Nevertheless, the documents fetched using the `GET /documents` route will have those fields, this setting only affects the document hits upon search.

This list can be restricted to a selected set of attributes in the settings.
```

#### Example

Suppose you manage a database that contains information about movies. By adding the following settings, documents returned upon search will contain the fields `title`, `description`, `poster`, `release_date` and `rank`.

```bash
$ curl \
  -X POST 'http://localhost:7700/indexes/movies/settings' \
  --data '{
      "displayedAttributes": [
          "title",
          "description",
          "poster",
          "release_date",
          "rank"
      ]
  }'
```

## Data storing

> All fields are stored. **This behavior cannot be changed**.

Reviewer: Once we have accepted the PR of Thomas we should link it here.

Author: Noted!
